test/sonos.test.js

```diff
@@ -1,90 +1,86 @@
+        'AddURIToQueueResponse',
+        'AVTransport',
+        '<FirstTrackNumberEnqueued>1</FirstTrackNumberEnqueued><NewQueueLength>1</NewQueueLength><NumTracksAdded>1</NumTracksAdded>'
+      )
+      mockRequest('/MediaRenderer/AVTransport/Control',
+        '"urn:schemas-upnp-org:service:AVTransport:1#Seek"',
+        '<u:Seek xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><InstanceID>0</InstanceID><Unit>TRACK_NR</Unit><Target>1</Target></u:Seek>',
+        'SeekResponse',
+        'AVTransport'
+      )
+      mockRequest('/MediaRenderer/AVTransport/Control',
+        '"urn:schemas-upnp-org:service:AVTransport:1#Play"',
+        '<u:Play xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><InstanceID>0</InstanceID><Speed>1</Speed></u:Play>',
+        'PlayResponse',
+        'AVTransport'
+      )
       var sonos = new Sonos('localhost', 1400)
 
-      var actionCounter = 0
-
-      sonos.request = function (endpoint, action, body, _, callback) {
-        actionCounter++
-
-        if (actionCounter === 1) {
-          assert(endpoint === '/MediaRenderer/AVTransport/Control')
-          assert(action === '"urn:schemas-upnp-org:service:AVTransport:1#AddURIToQueue"')
-          assert(body === '<u:AddURIToQueue xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><InstanceID>0</InstanceID><EnqueuedURI>http://livingears.com/music/SceneNotHeard/091909/Do You Mind Kyla.mp3</EnqueuedURI><EnqueuedURIMetaData></EnqueuedURIMetaData><DesiredFirstTrackNumberEnqueued>0</DesiredFirstTrackNumberEnqueued><EnqueueAsNext>1</EnqueueAsNext></u:AddURIToQueue>')
-          callback(null, [{
-            FirstTrackNumberEnqueued: ['1'],
-            NewQueueLength: ['1'],
-            NumTracksAdded: ['1']
-          }])
-        } else if (actionCounter === 2) {
-          assert(endpoint === '/MediaRenderer/AVTransport/Control')
-          assert(action === '"urn:schemas-upnp-org:service:AVTransport:1#Seek"')
-          assert(body === '<u:Seek xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><InstanceID>0</InstanceID><Unit>TRACK_NR</Unit><Target>1</Target></u:Seek>')
-          callback(null, [{
-            $: {'xmlns:u': 'urn:schemas-upnp-org:service:AVTransport:1'}
-          }])
-        } else if (actionCounter === 3) {
-          assert(endpoint === '/MediaRenderer/AVTransport/Control')
-          assert(action === '"urn:schemas-upnp-org:service:AVTransport:1#Play"')
-          assert(body === '<u:Play xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><InstanceID>0</InstanceID><Speed>1</Speed></u:Play>')
-          done()
-        }
-      }
-
-      sonos.play('http://livingears.com/music/SceneNotHeard/091909/Do You Mind Kyla.mp3')
+      return sonos.play('http://livingears.com/music/SceneNotHeard/091909/Do You Mind Kyla.mp3')
```

Comment:
Site's unavailable trying to checkout your tunes :P https://archive.org/details/audio might be a better resource for examples.

Reply:
I hear you. Changed the sound.